dbms/src/Operators/UnorderedSourceOp.h

```diff
@@ -26,6 +26,7 @@ namespace DB
 /// Read blocks asyncly from Storage Layer by using read thread,
 /// The result can not guarantee the keep_order property
 class UnorderedSourceOp : public SourceOp
+    , public Awaitable
```

Comment:
Is inheriting from multiple classes aims at avoiding virtual function call?(I mean why not add a virtual function `isAwaitable()`)

Reply:
Yes, this implementation is just an option. 
Adding `isAwaitable()` means that the awaitable operator needs to override two methods.

Reply:
updated to ·`isAwaitable()`